rotkehlchen/accounting/events.py

```diff
@@ -247,15 +247,28 @@ def add_buy_and_corresponding_sell(
         - RemoteError if there is a problem reaching the price oracle server
         or with reading the response returned by the server
         """
+        # In the case where the fee asset is nominated in the asset that we receive
+        # take care that the fee is processed once we have the received asset in possesion.
+        if bought_asset == fee_currency:
```

Comment:
I really don't understand what you are doing here or why. Same in `add_sell_and_corresponding_buy`

Reply:
What this does is to ensure that if the fee asset is the same as the asset received the fee is processed once we have the asset. So what it does is to choose where to put the fee, if in the sell part or the buy part of the trade